### PR TITLE
Add mud bricks to paths and sanity-fix the dyer

### PR DESCRIPTION
**Pull request by uecasm**
Closes [discord request](https://discord.com/channels/472875599422291968/486940969737125922/1221891519792021604)
Closes [discord request](https://discord.com/channels/139070364159311872/724810019928801290/1225913588485718162)
Closes [discord request](https://discord.com/channels/472875599422291968/1266391713509806184/1266391713509806184)

# Changes proposed in this pull request:
- Port #9288
- Move a bunch of DO ingredients from mechanic to sawmill or stonemason, to be more consistent with regular recipes.
- Fix mechanic sometimes duplicating DO ingredients handled by other crafters.
- Teach the dyer to obey guild rules
    - Remove various duplicated crafting items already handled by other crafters (e.g. beds from wool, glass panes from glass).
    - They can still craft those items, but only with the recipes that use dye.
- Allow postbox requests for dyed leather armor (if you have an example in your inventory).
- Show example dyed leather armor in the dyer's JEI info.
    image
- Add path block auditor.  (This was already visible in the block tags audit, but it's easier to read this way.)

[x] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

**Review by Raycoms — DISMISSED** [review]

**Comment by someaddons** [comment]
while on this we probably should also add the missing cracked variants for e.g. stone brick

File: src/datagen/generated/minecolonies/data/minecolonies/tags/blocks/pathblocks.json [hunk 1]
```diff
@@ -28,6 +28,9 @@
     "minecraft:chiseled_nether_bricks",
     "minecraft:gravel",
     "minecraft:dirt_path",
+    "minecraft:mud_bricks",
```
Comment (Raycoms) [inline src/datagen/generated/minecolonies/data/minecolonies/tags/blocks/pathblocks.json:31]: Can you already add the missing strong brick, stone brick stairs, cracked etc?

Comment (uecasm) [reply src/datagen/generated/minecolonies/data/minecolonies/tags/blocks/pathblocks.json:31]: Cracked stone bricks are already in there via the STONE_BRICKS tag.

Comment (uecasm) [reply src/datagen/generated/minecolonies/data/minecolonies/tags/blocks/pathblocks.json:31]: I've added a [path tag audit](https://docs.google.com/spreadsheets/d/1i6EQuPuxVWDmH2Z9_CHiXOmDG5sXVF2cscHg6lwXs9o/edit?gid=1294272618#gid=1294272618) now, which makes it easier to see which blocks are paths and which aren't.

**Review by Raycoms — APPROVED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**